File: mobile/lib/models/achievement.dart
class Achievement {
  final String title;
  final String description;
  final String iconUrl;
  final bool isUnlocked;

  Achievement({
    required this.title,
    required this.description,
    required this.iconUrl,
    required this.isUnlocked,
  });

  factory Achievement.fromJson(Map<String, dynamic> json) => Achievement(
    title: json['title'],
    description: json['description'],
    iconUrl: json['iconUrl'],
    isUnlocked: json['isUnlocked'],
  );
}

File: safe-road-backend/module-gamification/src/main/java/ru/itmo/saferoad/gamification/domain/Achievement.java
package ru.itmo.saferoad.gamification.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.NonNull;
import lombok.Setter;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

@Getter
@Setter
@Entity
@NoArgsConstructor
@Table(name = "\"Achievements\"")
@EqualsAndHashCode(of = "id")
public class Achievement {

	@Id
	@NonNull
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;

	@NonNull
	@Column(name = "name", nullable = false, unique = true)
	private String name;

	@NonNull
	@Column(name = "description", nullable = false)
	private String description;

	@NonNull
	@Column(name = "icon_url", nullable = false)
	private String iconUrl;

	@NonNull
	@JdbcTypeCode(SqlTypes.JSON)
	@Column(name = "requirements", nullable = false)
	private AchievementRequirements requirements;

	@NonNull
	@Column(name = "reward_xp", nullable = false)
	private Integer rewardXp;

	@NonNull
	@Column(name = "is_active", nullable = false)
	private Boolean isActive;
}
